### ISSUE TITLE
Using matlab project to create a project is a huge mistake！

### ISSUE THREAD

**Issue by cuixing158** [MathWorks-Teaching-Resources/Numerical-Methods-with-Applications#1]
As with Visual Studio project work, matlab has introduced [project](https://www.mathworks.com/help/matlab/projects.html) since 2019a, but I think this is an ill-advised move, as it makes it easy to organize folders in current projects, but it also creates problems. matlab should behave as "directly available "It is true that fewer and fewer customers are using project in their own repo, and that is why it is not widely accepted, judging from the open source projects of the last few years!

**Comment by eszmw** [comment]
It is convenient but not a requirement to begin by loading the project file in this repository. Once you have cloned this repository and opened it in MATLAB you could ignore NumericalMethods.prj and start by adding the relevant folders to the path and then running OpenOverview.m as follows:

```
>> addpath("NumericalPDEs")
>> addpath("NumericalODEs")
>> addpath("NumericalIntegration")
>> addpath("Interpolation")
>> addpath("HelperFunctions")
>> addpath("Data")
>> addpath("Derivatives")
>> OpenOverview
```

When you are done with the scripts, you can remove them from your path using

```
>> rmpath("HelperFunctions") 
>> rmpath("Derivatives")
>> rmpath("Data")
>> rmpath("Interpolation")
>> rmpath("NumericalPDEs")
>> rmpath("NumericalODEs")
>> rmpath("NumericalIntegration")
```

Can you explain what problems you have encountered due to projects?

**Comment by cuixing158** [comment]
@eszmw What I mean is that it is not the use of the function that is problematic, but the introduction of the [project](https://www.mathworks.com/help/matlab/projects.html) function that is "be completely superfluous". matlab is a high-level computing language that is directly problem-oriented, writing programs to obtain output as it goes, and there is absolutely no need to create a project framework like Visual C++ Project. It may seem necessary to organise files for "big projects", but matlab is widely used and satisfied with "short and concise" code for the vast majority of tasks (if not, TMW will consider introducing a new buildin function integration), project needs to consider the configuration of the environment in addition to providing the code files, which can add to the overhead of removing the problems themselves and cause discomfort to the user, both during creation and during use.

我的意思是：不是函数使用上出现问题，而是引入这个[project](https://www.mathworks.com/help/matlab/projects.html)功能就是“画蛇添足”。matlab是一种高级计算语言，直接面向问题编程，随即写程序随即获得输出结果，完全没必要像C++那样搞一个project工程框架出来，看似是为“大项目”组织文件所需，但matlab在面向绝大数任务时候，“短少精简”的代码是广泛使用和满足的（如果不满足，TMW会考虑引进一个新的buildin-function集成），project除了提供代码文件，还需要考虑配置环境，无论是创建还是使用过程中，会增加除去本身问题的开销，引起使用者的不适。

**Comment by eszmw** [comment]
I respectfully disagree. Projects work quite well for creating a collection of scripts, models, and supporting files that can be shared with an easily reproducible environment. They are certainly overkill for many uses of MATLAB, but for the creation of courseware I find them invaluable. In R2025a and later, the support for projects is even better. In addition to the path controls I referenced earlier, the dependency analyzer and Check Project functions help me keep my files organized and notify me if I accidentally add or delete files. Using projects allows me to create standard startup and cleanup experiences and helps with CI.
